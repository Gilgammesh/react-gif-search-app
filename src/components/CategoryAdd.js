import React, { useState } from "react";
import PropTypes from "prop-types";

const CategoryAdd = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = ({ target: { value } }) => {
    setInputValue(value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategorias((categ) => [inputValue, ...categ]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        name="categoria"
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

CategoryAdd.propTypes = {
  setCategorias: PropTypes.func.isRequired,
};

export default CategoryAdd;
