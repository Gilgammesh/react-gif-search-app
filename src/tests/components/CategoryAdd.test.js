import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import CategoryAdd from "../../components/CategoryAdd";

describe("Pruebas unitarias en el componente CategoryAdd.js", () => {
  const func = () => {};
  /************************************************************************************/
  test("se debe mostrar el componente <CategoryAdd /> correctamente", () => {
    const wrapper = shallow(<CategoryAdd setCategorias={func} />);
    expect(wrapper).toMatchSnapshot();
  });
  /************************************************************************************/
  test("se debe cambiar la caja texto", () => {
    const value = "naruto";
    const wrapper = shallow(<CategoryAdd setCategorias={func} />);
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });
    const input_ = wrapper.find("input"); // despues del cambio capturo nuevamente el input
    expect(input_.props().value).toBe(value);
  });
});
