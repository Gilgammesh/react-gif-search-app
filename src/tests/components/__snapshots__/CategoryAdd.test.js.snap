// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente CategoryAdd.js se debe mostrar el componente <CategoryAdd /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="categoria"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
